app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,56 @@
+class TasksController < ApplicationController
+
+  def index
+    @tasks = Task.order(:completion_date)
+  end
+
+  def show
+    @task = Task.find(params[:id].to_i)
+  end
+
+  def edit
+    @task = Task.find(params[:id].to_i)
+    unless @task
+      redirect_to books_path
+    end
+  end
+
+  def mark_complete
+    task = Task.find(params[:id].to_i)
+    redirect_to tasks_path unless task
+    status = (task.status == "incomplete" ? "complete":"incomplete")
+    task.update(status: status, completion_date: Date.today())
+    redirect_to root_path
+  end
+
+  def update
+    @task = Task.find(params[:id] )
+
+    redirect_to tasks_path unless @task
+    @task.name = params[:task][:name]
+    if @task.update_attributes params.require(:task).permit(:name, :completion_date, :description)
```

Comment:
Good use of strong params, but having a helper method do it makes things more readable and brief.